tools/dependency/validate.py

```diff
@@ -188,7 +203,8 @@ def ValidateControlPlaneDeps(self):
     # these paths.
     queried_controlplane_core_min_deps = self._build_graph.QueryExternalDeps(
         '//source/common/config/...')
-    expected_controlplane_core_deps = self._dep_info.DepsByUseCategory('controlplane')
+    expected_controlplane_core_deps = self._dep_info.DepsByUseCategory('controlplane').union(
```

Comment:
just to clarify: it seems expected controlplane deps will be covered by controlplane and api so it's not really a problem they're entangled like dataplane?

Reply:
Yeah, not a ton of value in disentangling these, whereas on dataplane, we do want to be clear about what is marginally contribute by certain extensions. Will add a comment.